roxie/ccd/ccdserver.cpp

```diff
@@ -2258,10 +2259,17 @@ class RecordPullerThread : public RestartableThread
                 inputStream->stop();
         }
         RestartableThread::join();
+        stopped = true;
     }
 
     void reset()
     {
+        if (!stopped)
```

Comment:
This code should not be needed if all other code in roxie was perfect - but it guards against resetting the puller without having first stopped it.